### PR TITLE
Add ironiapi tests

### PR DESCRIPTION
**Pull request by hjensas**

### CODE REVIEW

**Comment by hjensas** [comment]
/test ironic-operator-build-deploy-kuttl
/test ironic-operator-build-deploy-tempest

Issue with RPM repos, not related to this change:

`Errors during downloading metadata for repository 'localdev-rhel-8-server-ose-rpms'`
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_ironic-operator/313/pull-ci-openstack-k8s-operators-ironic-operator-main-ironic-operator-build-deploy-kuttl/1687425423689912320/artifacts/build-logs/openstack-operator-ci-build-deploy-kuttl.log

File: tests/functional/ironicapi_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2022 Red Hat Inc.
```
Comment (juliakreger) [inline tests/functional/ironicapi_controller_test.go:2]: 2023 :)

Comment (hjensas) [reply tests/functional/ironicapi_controller_test.go]: oops, I rebased and fixed this nit in the last push.

**Review by steveb — APPROVED** [review]
/lgtm

**Pull request merged**